python-squid/src/test/java/org/sonar/python/semantic/SymbolTableBuilderTreeTest.java

```diff
@@ -149,6 +149,26 @@ public void lambdas() {
     assertThat(symbolByName.keySet()).containsOnly("z");
   }
 
+  @Test
+  public void for_stmt() {
+    PyFunctionDefTree functionTree = functionTreesByName.get("function_with_loops");
+    Map<String, TreeSymbol> symbolByName = functionTree.localVariables().stream().collect(Collectors.toMap(TreeSymbol::name, Functions.identity()));
+
+    assertThat(symbolByName.keySet()).containsOnly("x");
+    TreeSymbol x = symbolByName.get("x");
+    assertThat(x.usages()).hasSize(1);
+  }
+
+  @Test
+  public void comprehension() {
+    PyFunctionDefTree functionTree = functionTreesByName.get("function_with_comprehension");
+    Map<String, TreeSymbol> symbolByName = functionTree.localVariables().stream().collect(Collectors.toMap(TreeSymbol::name, Functions.identity()));
+
+    assertThat(symbolByName.keySet()).containsOnly("a");
```

Comment:
I think it's wrong: I think `a` is a local variable of the comprehension, not a local variable of the function.

Reply:
It looks like there is a difference between Python2 and Python3 scoping:
http://python-history.blogspot.com/2010/06/from-list-comprehensions-to-generator.html
> We also made another change in Python 3, to improve equivalence between list comprehensions and generator expressions. In Python 2, the list comprehension "leaks" the loop control variable into the surrounding scope:
>```
>x = 'before'
>a = [x for x in 1, 2, 3]
>print x # this prints '3', not 'before'
>```

I suggest to create a separate ticket for it